tests/models/test_case_small_network_optional_assets_annualized/src/run_annualized.py

```diff
@@ -0,0 +1,231 @@
+from typing import Any, Dict, List
+
+import numpy as np
+
+
+from rtctools.data.storage import DataStore
+from rtctools.optimization.collocated_integrated_optimization_problem import (
+    CollocatedIntegratedOptimizationProblem,
+)
+from rtctools.optimization.goal_programming_mixin import Goal, GoalProgrammingMixin
+from rtctools.optimization.linearized_order_goal_programming_mixin import (
+    LinearizedOrderGoalProgrammingMixin,
+)
+from rtctools.util import run_optimization_problem
+
+from rtctools_heat_network.esdl.esdl_mixin import ESDLMixin
+from rtctools_heat_network.heat_mixin import HeatMixin
+
+
+class TargetDemandGoal(Goal):
+    priority = 1
+    order = 2
+
+    def __init__(self, state, target):
+        self.state = state
+
+        self.target_min = target
+        self.target_max = target
+        self.function_range = (-1.0, 2.0 * max(target.values))
+        self.function_nominal = np.median(target.values)
+
+    def function(self, optimization_problem, ensemble_member):
+        return optimization_problem.state(self.state)
+
+
+class MinimizeDiscAnnualizedCostGoal(Goal):
+
+    """This class represents an optimization goal that minimizes the
+      discounted annualized cost of a heat network model.
+
+    It takes the annualized cost calculated by the model and minimizes
+    it by changing optimization variables such as asset sizes.
+    """
+
+    order = 1
+    priority = 2
+
+    def __init__(self):
+        self.target_max = 0.0
+        self.function_range = (0.0, 1.0e8)
+        self.function_nominal = 1.0e7
+
+    def function(self, optimization_problem: HeatMixin, ensemble_member):
+        obj = 0.0
+
+        assets_and_cost_keys = [
+            (["source", "ates"], ["_asset_variable_operational_cost_map"]),
+            (["source", "ates", "buffer"], ["_asset_fixed_operational_cost_map"]),
+            (
+                [
+                    "source",
+                    "ates",
+                    "buffer",
+                    "demand",
+                    "heat_exchanger",
+                    "heat_pump",
+                    "pipe",
+                ],
+                ["_annualized_capex_var_map"],
+            ),
+        ]
```

Comment:
Can you move this code of the assets_and-cost_keys to the __init__ and make it configurable for the user?